src/NHapi.Base/Parser/Escape.cs

```diff
@@ -252,7 +269,7 @@ private static EncodingLookups BuildEncodingLookups(EncodingCharacters encChars)
             if (!VariousEncChars.ContainsKey(encChars))
             {
                 // this means we haven't got the sequences for these encoding characters yet - let's make them
-                VariousEncChars[encChars] = new EncodingLookups(encChars);
+                VariousEncChars.TryAdd(encChars, new EncodingLookups(encChars));
             }
```

Comment:
Is the `if` block still needed now that `VariousEncChars` is a concurrent dictionary?

Reply:
It's not strictly needed but it without it, the  `EncodingLoopups` is new'd up every time even when not needed, this means a new memory allocation for a new instance of  `EncodingLoopups` and garbage collection cycle on every usage.